operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/ManagedInformerEventSource.java

```diff
@@ -13,15 +18,31 @@
     extends CachingEventSource<R, P>
     implements ResourceEventHandler<R> {
 
+  @SuppressWarnings("rawtypes")
+  private static final ConcurrentMap<ResourceConfigurationAsKey, InformerManager> managedInformers =
+      new ConcurrentHashMap<>();
+
   protected ManagedInformerEventSource(
       MixedOperation<R, KubernetesResourceList<R>, Resource<R>> client, C configuration) {
     super(configuration.getResourceClass());
+    initCache(configuration);
     manager().initSources(client, configuration, this);
```

Comment:
Won't this multiple time init sources?

Reply:
Ehh we need a quite good quality intergation tests here IMHO. This is quite risky to do this way.